lua/dap/ext/vscode.lua

```diff
@@ -16,15 +16,29 @@ function M.load_launchjs(path)
   end
   local contents = table.concat(lines, '\n')
   local data = vim.fn.json_decode(contents)
+
+  -- For each adapter, groups debugees if they use the adapter itself.
+  -- { <adapter> = { <debugee> = boolean } }
+  local debugees_of_adapter = {}
+  for debugee, configs in pairs(dap.configurations) do
+    for _, config in ipairs(configs) do
+      local debugees = debugees_of_adapter[config.type] or {}
+      debugees_of_adapter[config.type] = debugees
+      debugees[debugee] = true
+    end
+  end
+
   assert(data.configurations, "launch.json must have a 'configurations' key")
   for _, config in ipairs(data.configurations) do
     assert(config.type, "Configuration in launch.json must have a 'type' key")
-    local configurations = dap.configurations[config.type]
-    if not configurations then
-      configurations = {}
-      dap.configurations[config.type] = configurations
+    local debugees = debugees_of_adapter[config.type]
+    if debugees then
```

Comment:
Doesn't this mean that if `dap.configurations` is initially empty,  nothing would be added? 

Did you see the conversations in https://github.com/mfussenegger/nvim-dap/pull/156 ?



The debugger naming is also a bit confusing to me - the introduction in dap.txt refers to the application being debugged as debugee but in this PR it's used for the configuration indices

Reply:
Oh, sorry. I didn't notice that PR. Also, It should be filetype not debugee itself, that's my fault.  

For the improvement, we can fallback to the old way (using adapter's name as filetype to add new debugee config) if there is no debugee config for that filetype. Doing it this way we need not to add any new mapping key between launch.json and nvim-dap.  What do you think?

Reply:
> For the improvement, we can fallback to the old way (using adapter's name as filetype to add new debugee config) if there is no debugee config for that filetype. Doing it this way we need not to add any new mapping key between launch.json and nvim-dap. What do you think?

Sounds good

Reply:
Just fixed it. Let me know if anything need to improve 🙂